### PR TITLE
Add unit test data for 1.14, 1.15 and 1.16

### PR DESCRIPTION
**Pull request by sayboras**
### Description

Fixes https://github.com/weaveworks/eksctl/issues/2178

**Note**: these config files are extracted from existing cluster for respective version. Some minor changes in version (e.g. coredns, cni, etc) as I might upgrade them along the way. Currently, the default versions for all versions are same as per https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html

### Testing

<details>
<summary>Create cluster</summary>

```shell scripts
$ ./eksctl create cluster --managed --node-private-networking --version 1.16
[ℹ]  eksctl version 0.20.0-dev+880af000.2020-05-14T22:40:08Z
[ℹ]  using region us-east-2
[ℹ]  setting availability zones to [us-east-2a us-east-2c us-east-2b]
[ℹ]  subnets for us-east-2a - public:192.168.0.0/19 private:192.168.96.0/19
[ℹ]  subnets for us-east-2c - public:192.168.32.0/19 private:192.168.128.0/19
[ℹ]  subnets for us-east-2b - public:192.168.64.0/19 private:192.168.160.0/19
[ℹ]  using Kubernetes version 1.16
[ℹ]  creating EKS cluster "floral-wardrobe-1589460163" in "us-east-2" region with managed nodes
[ℹ]  will create 2 separate CloudFormation stacks for cluster itself and the initial managed nodegroup
[ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=us-east-2 --cluster=floral-wardrobe-1589460163'
[ℹ]  CloudWatch logging will not be enabled for cluster "floral-wardrobe-1589460163" in "us-east-2"
[ℹ]  you can enable it with 'eksctl utils update-cluster-logging --region=us-east-2 --cluster=floral-wardrobe-1589460163'
[ℹ]  Kubernetes API endpoint access will use default of {publicAccess=true, privateAccess=false} for cluster "floral-wardrobe-1589460163" in "us-east-2"
[ℹ]  2 sequential tasks: { create cluster control plane "floral-wardrobe-1589460163", 2 parallel sub-tasks: { no tasks, create managed nodegroup "ng-c8e176f4" } }
[ℹ]  building cluster stack "eksctl-floral-wardrobe-1589460163-cluster"
[ℹ]  deploying stack "eksctl-floral-wardrobe-1589460163-cluster"
[ℹ]  building managed nodegroup stack "eksctl-floral-wardrobe-1589460163-nodegroup-ng-c8e176f4"
[ℹ]  deploying stack "eksctl-floral-wardrobe-1589460163-nodegroup-ng-c8e176f4"
[ℹ]  waiting for the control plane availability...
[✔]  saved kubeconfig as "/home/tammach/.kube/config"
[ℹ]  no tasks
[✔]  all EKS cluster resources for "floral-wardrobe-1589460163" have been created
[ℹ]  nodegroup "ng-c8e176f4" has 2 node(s)
[ℹ]  node "ip-192-168-102-156.us-east-2.compute.internal" is ready
[ℹ]  node "ip-192-168-154-154.us-east-2.compute.internal" is ready
[ℹ]  waiting for at least 2 node(s) to become ready in "ng-c8e176f4"
[ℹ]  nodegroup "ng-c8e176f4" has 2 node(s)
[ℹ]  node "ip-192-168-102-156.us-east-2.compute.internal" is ready
[ℹ]  node "ip-192-168-154-154.us-east-2.compute.internal" is ready
[ℹ]  kubectl command should work with "/home/tammach/.kube/config", try 'kubectl get nodes'
[✔]  EKS cluster "floral-wardrobe-1589460163" in "us-east-2" region is ready

$ kgno
NAME                                            STATUS   ROLES    AGE     VERSION
ip-192-168-102-156.us-east-2.compute.internal   Ready    <none>   4m39s   v1.16.8-eks-e16311
ip-192-168-154-154.us-east-2.compute.internal   Ready    <none>   4m46s   v1.16.8-eks-e16311

```

</details>

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: pkg/kubernetes/scheme.go [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 
 	"k8s.io/client-go/kubernetes/scheme"
 )
 
 func init() {
 	_ = apiextensionsv1beta1.AddToScheme(scheme.Scheme)
+	_ = apiextensionsv1.AddToScheme(scheme.Scheme)
```
Comment (sayboras) [inline pkg/kubernetes/scheme.go:12]: main changes related to kube 1.16 CRD

File: go.mod [hunk 2]
```diff
@@ -48,13 +49,13 @@ require (
 	github.com/whilp/git-urls v0.0.0-20160530060445-31bac0d230fa
 	golang.org/x/sys v0.0.0-20200428200454-593003d681fa // indirect
 	golang.org/x/tools v0.0.0-20200301222351-066e0c02454c
-	k8s.io/api v0.15.11
-	k8s.io/apiextensions-apiserver v0.15.11
-	k8s.io/apimachinery v0.15.11
-	k8s.io/cli-runtime v0.15.11
+	k8s.io/api v0.16.8
```
Comment (sayboras) [inline go.mod:52]: 1.16.8 is default version for eks 1.16 now

File: pkg/addons/default/aws_node_test.go [hunk 3]
```diff
@@ -46,15 +46,15 @@ var _ = Describe("default addons - aws-node", func() {
...
 			)
 
 		})
 
-		It("can update 1.12 sample to latest", func() {
+		It("can update 1.14 sample to latest", func() {
 			rawClient.AssumeObjectsMissing = false
 
-			_, err := UpdateAWSNode(rawClient, "eu-west-2", false)
+			_, err := UpdateAWSNode(rawClient, "eu-west-1", false)
```
Comment (michaelbeaumont) [inline pkg/addons/default/aws_node_test.go:57]: why this change?

Comment (sayboras) [reply pkg/addons/default/aws_node_test.go:57]: this is for new file sample-1.14.json, which is the new file as part of this PR. I can modify the testdata/sample-*.json to all use us-west-2 regions as well. What do you have in mind ?

File: go.mod [hunk 4]
```diff
@@ -70,27 +71,28 @@ replace (
 	// Override version since auto-detected one fails with GOPROXY
 	github.com/census-instrumentation/opencensus-proto => github.com/census-instrumentation/opencensus-proto v0.2.0
 	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.3.1
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v0.9.4
```
Comment (michaelbeaumont) [inline go.mod:74]: Is this automatically here or did you need to add it? Is it also because of GOPROXY?

Comment (sayboras) [reply go.mod:74]: @michaelbeaumont I added it manually to avoid below issue during build.

```
../../../../pkg/mod/k8s.io/component-base@v0.16.8/metrics/legacyregistry/registry.go:44:9: undefined: prometheus.InstrumentHandler
```

k8s.io/kops (v1.15.2 or v1.16.2) is still using old version of component-base, so if we replace component-base, we need to replace its dependency (e.g. prometheus/client_golang) as well.

Latest alpha version of kops fixes this issue, but I reckon that it's better to wait for official release.

Comment (michaelbeaumont) [reply go.mod:74]: oh man, golang dependency management :sweat_smile:

Comment (sayboras) [reply go.mod:74]: haha yeah agreed that transitive dependencies for replace directive could be better :100:

File: go.mod [hunk 5]
```diff
@@ -71,6 +71,8 @@ replace (
 	// Override version since auto-detected one fails with GOPROXY
 	github.com/census-instrumentation/opencensus-proto => github.com/census-instrumentation/opencensus-proto v0.2.0
 	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.3.1
+	// k8s.io/kops is still using old version of component-base
```
Comment (sayboras) [inline go.mod:74]: Thanks a lot :100:, I am quite bad at noting down these important points :)
